test/methods.js

```diff
@@ -1282,4 +1314,25 @@ test("cpfBR", function() {
 	ok( !method( "11144477737"), "Invalid CPF Number: 2nd check number failed");
 });
 
+test("file accept - image wildcard", function() {
+	var input = acceptFileDummyInput("test.png", "image/png"),
+			$form = $("<form />"),
+			proxy = $.proxy($.validator.methods.accept, new $.validator({}, $form[0]), null, input, "image/*");
+	ok( proxy(), "the selected file for upload is valid" );
+});
+
+test("file accept - specified mime type", function() {
+	var input = acceptFileDummyInput("test.kml", "application/vnd.google-earth.kml+xml"),
+			$form = $("<form />"),
+			proxy = $.proxy($.validator.methods.accept, new $.validator({}, $form[0]), null, input, "application/vnd.google-earth.kml+xml");
+	ok( proxy(), "the selected file for upload is valid" );
+});
+
+test("file accept - invalid mime type", function() {
+	var input = acceptFileDummyInput("test.kml", "foobar/vnd.google-earth.kml+xml"),
+			$form = $("<form />"),
+			proxy = $.proxy($.validator.methods.accept, new $.validator({}, $form[0]), null, input, "application/vnd.google-earth.kml+xml");
+	equal(proxy(), "dependency-mismatch", "the selected file for upload is valid" );
```

Comment:
The return value `"dependency-mismatch"` comes from the `optional()` method, which indicates that the code in question doesn't actually, since that returns `false`. So it looks to me like the test is expecting the wrong result, and that there is an issue elsewhere causing the wrong code to run. It might be a problem with the test, it might be in the implementation.

Much less critical, the third argument, "the selected file for upload is valid", is wrong, at least based on what the test name says ("invalid mime type").